pyccel/stdlib/ndarrays/ndarrays.h

```diff
@@ -0,0 +1,68 @@
+#ifndef NDARRAYS_H
+# define NDARRAYS_H
+
+# include <stdlib.h>
+# include <complex.h>
+# include <string.h>
+# include <stdio.h>
+# include <stdarg.h>
+# include <stdbool.h> 
+typedef struct  s_slice_data
+{
+    int start;
+    int end;
+    int step;
+}               t_slice;
+
+enum e_types
+{
+        nd_int,
+        nd_float,
+        nd_double,
+        nd_cdouble
+};
+
+typedef struct  s_ndarray
+{
+    /* raw data buffer*/
+    union {
+            char            *raw_data;
+            int             *nd_int;
+            float           *nd_float;
+            double          *nd_double;
+            double complex  *nd_cdouble;
+            };
+    /* number of dimmensions */
+    int             nd;
+    /* shape 'size of each dimmension' */
+    int             *shape;
+    /* strides 'number of bytes to skip to get the next element' */
+    int             *strides;
+    /* type of the array elements */
+    enum e_types    type;
+    int             type_size;
+    int             length;
+    int             buffer_size;
+    bool            is_slice;
+}               t_ndarray;
+
+/* functions prototypes */
+
+/* allocations */
+t_ndarray   array_create(int nd, int *shape, enum e_types type);
+t_ndarray   array_init(char *temp, int nd, int *shape, enum e_types type);
+t_ndarray   array_ones(int nd, int *shape, enum e_types type);
+t_ndarray   array_zeros(int nd, int *shape, enum e_types type);
+// t_ndarray   array_full(void *c, int nd, int *shape, enum e_types type);
+
+/* slicing */
+t_slice     slice_data(int start, int end, int step);
+t_ndarray   slice_make(t_ndarray p, ...);

```

Comment:
I make this comment here, but it applies to multiple places in the code. As I mentioned at one of our meetings, there is some confusion with the use of the word "slice". In order to be consistent with Python, we should call **slice** only the `t_slice` object, and maybe the function that creates it. Hence:
```c
t_slice  slice(int start, int end, int step);  // Also OK to call it "new_slice" or "create_slice"
```
Regarding the action of **slicing** an array, it results in an array **view**. So I suggest we write
```c
t_ndarray  array_slicing(t_ndarray p, ...); // Also OK to call it "sliced_array"
```
Suggestions are welcome!

Reply:
i will go with new_slice 
and array_slicing "to match the other functions"